# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [4]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [3]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3         V4        V5         V6        V7       
1 0.1006315 0.4578952 0.09765357 0.8394551 0.03512344 0.0669491 0.2687691
2 0.2284193 0.3052366 0.91174464 0.8981354 0.80176065 0.8454781 0.5535637
  V8        V9        V10       V11        V12       V13       V14      
1 0.9277702 0.5540144 0.8406025 0.07350115 0.8451176 0.2990771 0.3023573
2 0.7953603 0.6510426 0.3053236 0.73117891 0.6926075 0.5757401 0.9688036
  V15       V16       V17       V18       V19       V20      
1 0.3381781 0.6530220 0.2658136 0.9230773 0.9205625 0.4783022
2 0.9430066 0.6914709 0.1960608 0.1374807 0.9826139 0.8638064

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>% gather(Key, Value)
  

head(df2)

Key Value     
1 V1  0.10063152
2 V1  0.22841932
3 V2  0.45789522
4 V2  0.30523664
5 V3  0.09765357
6 V3  0.91174464

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [7]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [8]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook